### PR TITLE
Extend typeDescToWords to handle all type descriptors

### LINKED ISSUE #555: Extend typeDescToWords to handle all type descriptors
`typeDescToWords` in `astString` should be extended to handle all type descriptors that can be parsed, not just the subset supported in functions, for example `type T true;`.

### PR DESCRIPTION
**Pull request by ushirask**
## Purpose
Fixes #555 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/front.syntax/astString.bal [hunk 1]
```diff
@@ -315,9 +313,35 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
...
-        panic err:impossible(`typedesc not implemented in typeDescToWords: ${td.toString()}`);
-    }   
+        w.push("function(", CLING);
+        boolean comma = false;
+        foreach var param in td.params {
+            if comma {
+                w.push(",");
+            }
+            typeDescToWords(w, param.td);
+            if param.name != null {
```
Comment (manuranga) [inline compiler/modules/front.syntax/astString.bal]:
```suggestion
            if param.name != () {
```

File: compiler/modules/front.syntax/astString.bal [hunk 2]
```diff
@@ -315,9 +313,35 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
...
+                w.push(param.name, CLING);
+            }
+            comma = true;
+        }
+        w.push(")");
+        TypeDesc ret = td.ret;
+        if !(ret is BuiltinTypeDesc && ret.builtinTypeName is "null") {
```
Comment (manuranga) [inline compiler/modules/front.syntax/astString.bal]: I prefer
```suggestion
        if ret !is BuiltinTypeDesc || ret.builtinTypeName != "null" {
```

File: compiler/modules/front.syntax/astString.bal [hunk 3]
```diff
@@ -315,9 +313,35 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
     else if td is SingletonTypeDesc {
         valueToWords(w, td.value);
     }
+    else if td is UnaryTypeDesc {
+        w.push(td.op, CLING);
+        typeDescToWords(w, td.td);
+    }
+    else if td is XmlSequenceTypeDesc {
+        w.push("xml<", CLING);
```
Comment (jclark) [inline compiler/modules/front.syntax/astString.bal]: Each token should be pushed separately, as elsewhere in this file i.e. `"xml", "<"`

File: compiler/modules/front.syntax/astString.bal [hunk 4]
```diff
@@ -315,9 +313,35 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
...
+        typeDescToWords(w, td.constituent);
+        w.push(CLING, ">");
+    }
     else {
-        panic err:impossible(`typedesc not implemented in typeDescToWords: ${td.toString()}`);
-    }   
+        w.push("function(", CLING);
```
Comment (jclark) [inline compiler/modules/front.syntax/astString.bal]: `"function", "("`

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/front.syntax/astString.bal (hunk 3 above)
Comment (ushirask) [reply compiler/modules/front.syntax/astString.bal]: Fixed.

**Pull request merged**